test/yugabyte_test.go

```diff
@@ -0,0 +1,282 @@
+		t.Fatalf("We got error while getting file from server :- Tserver: %s", getTserverFileError)
+	}
+
+	outputMasterConf := ssh.CheckSshCommand(t, sshSession, fmt.Sprintf("cat /home/%s/yugabyte-db/master/conf/server.conf", sshUser))
+	outputTserverConf := ssh.CheckSshCommand(t, sshSession, fmt.Sprintf("cat /home/%s/yugabyte-db/tserver/conf/server.conf", sshUser))
+	logger.Logf(t, "Master Conf:\n%s", outputMasterConf)
+	logger.Logf(t, "Tserver Conf:\n%s", outputTserverConf)
+}
+
+// Check Tserver process is running or not
+func testYugaByteProcess(t *testing.T, terraformOptions *terraform.Options, sshSession ssh.Host) {
+	// Enhancement - Verify Master Process status
+	// checkMasterProcess := "pgrep --list-name 'yb-master'"
+	checkTserverProcess := "pgrep --list-name 'yb-tserver'"
+	retry.DoWithRetry(t, "Checking process ", maxRetries, timeBetweenRetries, func() (string, error) {
+		// Enhancement - Verify Master Process status
+		// outputMasterProcess, errMasterProcess := ssh.CheckSshCommandE(t, sshSession, checkMasterProcess)
+		outputTserverProcess, errTserverProcess := ssh.CheckSshCommandE(t, sshSession, checkTserverProcess)
+		if errTserverProcess != nil {
+			return "", fmt.Errorf("Error Tserver Process: %s", errTserverProcess)
+		}
+		if !strings.Contains(string(outputTserverProcess), "yb-tserver") {
+			return "", fmt.Errorf("Process: %s", outputTserverProcess)
+		}
+		return "", nil
+	})
+}
+
+// Check Tserver logfile exists or not
+func testYugaByteLogFile(t *testing.T, terraformOptions *terraform.Options, sshUser string, sshSession ssh.Host) {
+	// Enhancement - Verify the Master Logfile exists or not
+	// logMasterDir := fmt.Sprintf("/home/%s/yugabyte-db/master/", sshUser)
+	logTserverDir := fmt.Sprintf("/home/%s/yugabyte-db/tserver/", sshUser)
+	retry.DoWithRetry(t, "Checking log files", maxRetries, timeBetweenRetries, func() (string, error) {
+		// Enhancement - Verify the Master Logfile exists or not
+		// outputMasterDir, errMasterDir := ssh.CheckSshCommandE(t, sshSession, fmt.Sprintf("ls %s | grep 'master.'", logMasterDir))
+		outputTserverDir, errTserverDir := ssh.CheckSshCommandE(t, sshSession, fmt.Sprintf("ls %s | grep 'tserver.'", logTserverDir))
+		if errTserverDir != nil {
+			return "", fmt.Errorf("Error Tserver Logfile: %s", errTserverDir)
+		}
+		if !strings.Contains(string(outputTserverDir), ".out") {
+			return "", fmt.Errorf("Log files: %s", outputTserverDir)
+		}
+		return "", nil
+	})
+}
+
+// Verify the Ulimit configuration of node
+func testYugaByteNodeUlimit(t *testing.T, terraformOptions *terraform.Options, sshSession ssh.Host) {
+	ulimitCommand := "ulimit -a"
+	retry.DoWithRetry(t, "Checking Ulimits", maxRetries, timeBetweenRetries, func() (string, error) {
+		output, err := ssh.CheckSshCommandE(t, sshSession, ulimitCommand)
+		if err != nil {
+			return "", err
+		}
+		if !strings.Contains(string(output), "119934") && !strings.Contains(string(output), "1048576") && !strings.Contains(string(output), "12000") {
```

Comment:
this seems like a slightly flimsy check. I suggest we remove the ulimit check entirely because it is verifying terraform functionality more than any yugabyte functionality.

Reply:
I thought we'll check both the functionalities of YugabyteDB as well as Terraform. We should check the terraform functionality because if something went wrong with the terraform deployment, this check will make it easier to catch that issue.